WooCommerce/Classes/ViewModels/Order Details/Refunded Products/RefundedProductsViewModel.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+import Yosemite
+
+
+// MARK: - View Model for the Refunded Products view controller
+//
+final class RefundedProductsViewModel {
+    /// Order we're observing.
+    ///
+    private(set) var order: Order
+
+    /// Aggregate data for all OrderItemRefund.
+    ///
+    private var refundedProducts: [AggregateOrderItem]
+
+    /// The datasource that will be used to render the Refunded Products screen.
+    ///
+    private(set) lazy var dataSource: RefundedProductsDataSource = {
+        let sortedItems = refundedProducts.sorted(by: { ($0.productID, $0.variationID) < ($1.productID, $1.variationID) })
+        return RefundedProductsDataSource(order: order, refundedProducts: sortedItems)
+    }()
+
+    /// Designated initializer.
+    ///
+    init(order: Order, refundedProducts: [AggregateOrderItem]) {
+        self.order = order
+        self.refundedProducts = refundedProducts
+    }
+
+    /// Update the view model's order when notified
+    ///
+    func update(order newOrder: Order) {
+        self.order = newOrder
```

Comment:
do you know if other properties need to be updated when the Order is updated?

Reply:
🤔 the OrderStatus is also updated, in `updateOrderStatus(order:)`, and is the function that calls this one. I think that's it.